src/test/java/com/learning/pages/swaglabs/SwagLabsLoginPageTest.java

```diff
@@ -0,0 +1,109 @@
+package com.learning.pages.swaglabs;
+
+import static org.testng.Assert.assertEquals;
+
+import com.google.inject.Inject;
+import com.learning.browsers.BrowserDriver;
+import com.learning.browsers.DriverModule;
+import com.learning.configuration.PropertiesReader;
+import org.openqa.selenium.WebDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Guice;
+import org.testng.annotations.Test;
+
+@Guice(modules = DriverModule.class)
+public class SwagLabsLoginPageTest {
+
+  @Inject
+  private BrowserDriver browserDriver;
+
+  @Inject
+  private PropertiesReader propertiesReader;
+
+  private WebDriver driver;
+  private SwagLabsLoginPage swagLabsLoginPage;
+
+  @BeforeClass
+  void setUp() {
+    driver = browserDriver.getDriver();
+    swagLabsLoginPage = new SwagLabsLoginPage(driver, propertiesReader);
+  }
+
+  @Test(dataProvider = "successfulLoginData")
+  void shouldGoToShopPageWhenSuccessfulLogin(String username, String password) {
+    //Given
+    String expectedPageUrl = "https://www.saucedemo.com/inventory.html";
+    //When
+    driver.get(swagLabsLoginPage.getSwagLabsURL());
+    waitForPageToLoad();
+    swagLabsLoginPage.performLogin(username, password);
+    String actualPageUrl = driver.getCurrentUrl();
+    //Then
+    assertEquals(actualPageUrl, expectedPageUrl);
+  }
+
+  @Test(dataProvider = "unsuccessfulLoginData")
+  void shouldErrorMessageBeVisibleWhenUnsuccessfulLogin(String username, String password) {
+    //Given
+    String expectedErrorMessage = "Epic sadface: Username and password do not match any user in this service";
+    //When
+    driver.get(swagLabsLoginPage.getSwagLabsURL());
+    swagLabsLoginPage.performLogin(username, password);
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
+    //Then
+    assertEquals(actualErrorMessage, expectedErrorMessage);
+  }
+
+  @Test(dataProvider = "unsuccessfulLoginData")
+  void shouldErrorMessageBeVisibleWhenLackOfPassword(String username, String password) {
+    //Given
+    String expectedErrorMessage = "Epic sadface: Password is required";
+    //When
+    driver.get(swagLabsLoginPage.getSwagLabsURL());
+    swagLabsLoginPage.performLogin(username, "");
+    String actualErrorMessage = swagLabsLoginPage.getErrorText();
+    //Then
+    assertEquals(actualErrorMessage, expectedErrorMessage);
+  }
+
+  @Test(dataProvider = "unsuccessfulLoginData")
+  void shouldErrorMessageBeVisibleWhenLackOfUsername(String username, String password) {
+    //Given
+    String expectedErrorMessage = "Epic sadface: Username is required";
+    //When
+    driver.get(swagLabsLoginPage.getSwagLabsURL());
+    waitForPageToLoad();
```

Comment:
Is this right usage of this method?

Reply:
Not exactly. There are still few that would have to be implemented, but it could look like this
```
browser.open(SwagLabsLoginPage.class)
```

Browser could be an wrapper on driver, hiding driver completely.
Benefits? No selenium classes in tests... just page objects and browser.

I'll try to modify this branch to show you how this can be done.